Panic on requests with private key auth when DPoP is disabled

### Describe the bug?

With SDK version 4.1.0, every second call to API results in panic.

The root cause: https://github.com/okta/okta-sdk-golang/pull/454/files#diff-9f03089e1a25ed798b9db4fd587bb8a9a4f94eca8df0234aef9ac2b6f298b02cR197

in `Authorize`, nil value for key is received from `getAccessTokenForPrivateKey` (I assume, that only happens when dpop is disabled, which is my case). This value is then stored in cache, just like the empty `nonce`.

On next call, nil `privateKey` value is [retrieved](https://github.com/okta/okta-sdk-golang/pull/454/files#diff-9f03089e1a25ed798b9db4fd587bb8a9a4f94eca8df0234aef9ac2b6f298b02cR282) from cache and provided to `generateDpopJWT`, which causes code to panic.

So, there are two issues actually:
- SDK should be aware that dpop is optional and not panic
- Cache should check empty/nil values (or they simply shouldn't be stored there at least)

### What is expected to happen?

Subsequent calls to API methods work, just like in previous version

### What is the actual behavior?

panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1016886d8]

goroutine 57 [running]:
github.com/okta/okta-sdk-golang/v4/okta.generateDpopJWT(0x0, {0x1030071c0, 0x3}, {0x1400026e180, 0x25}, {0x0, 0x0}, {0x14000c86387, 0x34e})

### Reproduction Steps?

- call any function
- call any function again

In my case it was users listing with pagination that broke everything.

### Additional Information?

_No response_

### Golang Version

go version go1.21.3 darwin/arm64

### SDK Version

v4.1.0

### OS version

_No response_

--- a/.generator/config.yaml
+++ b/.generator/config.yaml
@@ -8,7 +8,7 @@ additionalProperties:
   enumClassPrefix: true
   generateInterfaces: true
   packageName: okta
-  packageVersion: 4.1.0
+  packageVersion: 4.1.1
   useOneOfDiscriminatorLookup: true
   disallowAdditionalPropertiesIfNotPresent: false
 files:

--- a/.generator/okta-management-APIs-oasv3-noEnums-inheritance.yaml
+++ b/.generator/okta-management-APIs-oasv3-noEnums-inheritance.yaml
@@ -33112,9 +33112,7 @@ components:
           readOnly: true
         detailEntry:
           type: object
-          additionalProperties:
-            type: object
-            properties: {}
+          additionalProperties: true
           readOnly: true
         displayName:
           type: string

--- a/.generator/templates/client.mustache
+++ b/.generator/templates/client.mustache
@@ -161,13 +161,13 @@ func NewPrivateKeyAuth(config PrivateKeyAuthConfig) *PrivateKeyAuth {
 
 func (a *PrivateKeyAuth) Authorize(method, URL string) error {
 	accessToken, hasToken := a.tokenCache.Get(AccessTokenCacheKey)
-	if hasToken {
+	if hasToken && accessToken != "" {
 		accessTokenWithTokenType := accessToken.(string)
 		a.req.Header.Add("Authorization", accessTokenWithTokenType)
 		nonce, hasNonce := a.tokenCache.Get(DpopAccessTokenNonce)
-		if hasNonce {
+		if hasNonce && nonce != "" {
 			privateKey, ok := a.tokenCache.Get(DpopAccessTokenPrivateKey)
-			if ok {
+			if ok && privateKey != nil {
 				res := strings.Split(accessTokenWithTokenType, " ")
 				if len(res) != 2 {
 					return errors.New("Unidentified access token")
@@ -178,6 +178,8 @@ func (a *PrivateKeyAuth) Authorize(method, URL string) error {
 				}
 				a.req.Header.Set("Dpop", dpopJWT)
 				a.req.Header.Set("x-okta-user-agent-extended", "isDPoP:true")
+			} else {
+				return errors.New("Using Dpop but signing key not found")
 			}
 		}
 	} else {
@@ -259,13 +261,13 @@ func NewJWTAuth(config JWTAuthConfig) *JWTAuth {
 
 func (a *JWTAuth) Authorize(method, URL string) error {
 	accessToken, hasToken := a.tokenCache.Get(AccessTokenCacheKey)
-	if hasToken {
+	if hasToken && accessToken != "" {
 		accessTokenWithTokenType := accessToken.(string)
 		a.req.Header.Add("Authorization", accessTokenWithTokenType)
 		nonce, hasNonce := a.tokenCache.Get(DpopAccessTokenNonce)
-		if hasNonce {
+		if hasNonce && nonce != "" {
 			privateKey, ok := a.tokenCache.Get(DpopAccessTokenPrivateKey)
-			if ok {
+			if ok && privateKey != nil {
 				res := strings.Split(accessTokenWithTokenType, " ")
 				if len(res) != 2 {
 					return errors.New("Unidentified access token")
@@ -276,6 +278,8 @@ func (a *JWTAuth) Authorize(method, URL string) error {
 				}
 				a.req.Header.Set("Dpop", dpopJWT)
 				a.req.Header.Set("x-okta-user-agent-extended", "isDPoP:true")
+			} else {
+				return errors.New("Using Dpop but signing key not found")
 			}
 		}
 	} else {

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 Running changelog of releases since `2.0.0-rc.4`
 
 ## v4.1.0
+ - Fix panic issue when using bearer token (#463) Thanks [@duytiennguyen-okta]
+ - Fix object that does not have additional properties (#463) Thanks [@duytiennguyen-okta]
+
+## v4.1.0
  - Add support for dpop (#454) Thanks [@duytiennguyen-okta]
  - Fix object that does not have additional properties (#456) Thanks [@duytiennguyen-okta]
  - Fix header accept issue that only accept application/json (#457) Thanks [@duytiennguyen-okta]

